source/jormungandr/jormungandr/ptref.py

```diff
@@ -58,3 +71,28 @@ def get_stop_point(self, line_uri, code_key, code_value):
 
         logging.getLogger(__name__).info('PtRef, Multiple stop_points found with filter {}'.format(req.ptref.filter))
         return None
+
+    def get_objs(self, pb_type, filter=''):
+        """
+        iterator on all navitia objects that match a filter
+        """
+        req = request_pb2.Request()
+        req.requested_api = type_pb2.PTREFERENTIAL
+        req.ptref.requested_type = pb_type
+        req.ptref.count = 20
+        req.ptref.start_page = 0
+        req.ptref.depth = 1
+        req.ptref.filter = filter
+
+        while True:
+            result = self.instance.send_and_receive(req)
+
+            objects = getattr(result, PT_TYPE_RESPONSE_MAPPING[pb_type])
+            if not objects:
```

Comment:
You could also check if your aren't on the last page, it will prevent a useless call